spec/models/user_spec.rb

```diff
@@ -24,6 +24,76 @@
 
   describe 'validations' do
     it { is_expected.to validate_presence_of(:email) }
-    it { is_expected.to validate_uniqueness_of(:email) }
+    it { is_expected.to validate_uniqueness_of(:email).case_insensitive }
+  end
+
+  describe '.from_omniauth' do
+    let(:auth_params) { OmniAuth::AuthHash.new(
+      provider: :psu,
+      uid: 'jd1',
+      info: {
+        uid: 'jd1',
+        access_id: 'jd1',
+        email: 'jd1@psu.edu',
+        admin_area: 'University Libraries',
+        first_name: 'Joe',
+        last_name: 'Developer',
+        primary_affiliation: 'STAFF',
+        groups: ['whatever']
+      }
+    )
+    }
+
+    context 'when the User record does not yet exist' do
+      it 'creates a new User record' do
+        expect { described_class.from_omniauth(auth_params) }
+          .to change(described_class, :count)
+          .by 1
+      end
+
+      it 'returns the newly created User' do
+        new_user = described_class.from_omniauth(auth_params)
+        expect(new_user).to be_persisted
+        expect(new_user.access_id).to eq 'jd1'
+        expect(new_user.name).to eq 'Joe Developer'
+        expect(new_user.email).to eq 'jd1@psu.edu'
+      end
+    end
+
+    context 'when the User record already exists' do
+      let!(:existing_user) { create :user, provider: auth_params.provider, uid: auth_params.uid }
+
+      it 'does NOT create a new record' do
+        expect { described_class.from_omniauth(auth_params) }.not_to change(described_class, :count)
+      end
+
+      it 'does NOT update ANY attributes on the user record' do
+        user_before = described_class.find(existing_user.id)
+        described_class.from_omniauth(auth_params)
+        user_after = described_class.find(existing_user.id)
+
+        expect(user_after.access_id).to eq user_before.access_id
+        expect(user_after.email).to eq user_before.email
+        expect(user_after.name).to eq user_before.name
+      end
+
+      it 'returns the User record' do
+        expect(described_class.from_omniauth(auth_params)).to eq existing_user
+      end
+    end
+  end
+
+  # Required to exist by Blacklight
+  describe '#bookmarks' do
+    subject { described_class.new.bookmarks }
+
+    it { is_expected.to eq [] }
+  end
+
+  # Blacklight uses this method in the navigation header. I don't like this
+  describe '#to_s' do
+    subject { described_class.new(access_id: 'abc123').to_s }
+
+    it { is_expected.to eq 'abc123' }
```

Comment:
@rschenk @whereismyjetpack we shouldn't need these two tests if just run the generator that adds the Blacklight user module. We could have a `expect(user).to respond_to(:bookmarks) }` 🤷‍♂